22892 range tick mark (#23135)
mdn/content: files/en-us/web/html/element/input/range/index.md

This value must be less than or equal to the value of the [`max`](/en-US/docs/Web/HTML/Attributes/max) attribute. See the HTML [`min`](/en-US/docs/Web/HTML/Attributes/min) attribute.

> **Note:** If the `min` and `max` values are equal or the `max` value is lower than the `min` value the user will not be able to interact with the range.

### step

The `step` attribute is a number that specifies the granularity that the value must adhere to, or the special value `any`, which is described below. Only values which are equal to the basis for stepping ([`min`](#min) if specified, {{htmlattrxref("value", "input")}} otherwise, and an appropriate default value if neither of those is provided) are valid.

> A string value of `any` means that no stepping is implied, and any value is allowed (barring other constraints, such as [`min`](#min) and [`max`](#max)). See the [Setting step to "any"](#setting_step_to_any) example for how this works in supported browsers.

Reviewer: ```suggestion
A string value of `any` means that no stepping is implied, and any value is allowed (barring other constraints, such as [`min`](#min) and [`max`](#max)). See the [Setting step to `any`](#setting_step_to_any) example for how this works in supported browsers.
```

> A string value of `any` means that no stepping is implied, and any value is allowed (barring other constraints, such as [`min`](#min) and [`max`](#max)). See the [Setting step to "any"](#setting_step_to_any) example for how this works in supported browsers.

Reviewer: ```suggestion
A string value of `any` means that no stepping is implied, and any value is allowed (barring other constraints, such as [`min`](#min) and [`max`](#max)). See the [Setting step to `any`](#setting_step_to_any) example for how this works in supported browsers.
```

> > **Note:** When the data entered by the user doesn't adhere to the stepping configuration, the {{Glossary("user agent")}} may round to the nearest valid value, preferring to round up numbers when there are two equally close options.

Reviewer: ```suggestion
> **Note:** When the data entered by the user doesn't adhere to the stepping configuration, the {{Glossary("user agent")}} may round to the nearest valid value, preferring to round numbers up when there are two equally close options.
```

Author: corrected these

### Setting the value's granularity

> By default, the granularity, is 1, meaning that the value is always an integer. You can change the {{htmlattrxref("step")}} attribute to control the granularity. For example, If you need a value between 5 and 10, accurate to two decimal places, you should set the value of `step` to 0.5:

Reviewer: original was correct for the sentence (accurate to 2 decimal places is 0.01, not 0.5)

Author: So I changed this from 0.01 as the efect in the demo was so difficult to distinguish, so I shall update the "accurate to two decimal places" statement instead.

Author: updated

#### Setting the step attribute

```html

> <input type="range" min="5" max="10" step="0.5" />

Reviewer: original was valid

Author: See comment - https://github.com/mdn/content/pull/23135#discussion_r1056168259

Author: updated

If you want to accept any value regardless of how many decimal places it extends to, you can specify a value of `any` for the {{htmlattrxref("step", "input")}} attribute:

##### HTML

```html
<input id="pi_input" type="range" min="0" max="3.14" step="any" />

> <p>Value: <span id="value">1.57</span></p>

Reviewer: I recommend using `<output>` instead of `span`, and letting the JS fill, not pre-fill, the value.

Author: Of course.

Author: updated

```

##### JavaScript

```js
const value = document.querySelector("#value")
const input = document.querySelector("#pi_input")
input.addEventListener("input", (event) => {
  value.textContent = event.target.value
})
```

{{EmbedLiveSample("Setting_step_to_any", 600, 75)}}

> This example lets the user select any value between 0 and π without any restriction on the fractional part of the value selected. The JavaScript is used to show how the value is changing as the range is interacted with.

Reviewer: ```suggestion
This example lets the user select any value between 0 and π without any restriction on the fractional part of the value selected. JavaScript is used to show how the value changes as the user interacts with the range.
```

#### Result

{{EmbedLiveSample("Adding tick marks", 600, 50)}}

### Using the same datalist for multiple range controls

> To help you from repeating code you can reuse that same {{HTMLElement("datalist")}} for multiple `<input type="range">` elements.

Reviewer: ```suggestion
To help you from repeating code you can reuse that same {{HTMLElement("datalist")}} for multiple `<input type="range">` elements, and other  {{HTMLElement("input")}} types.
```

> **Note:** If you also want to [show the labels](#adding_labels) as in the example below then you would need a `datalist` for each range input.

#### HTML

```html
<label for="temp">Temperature for room 1:</label><br />
<input type="range" id="temp" name="temp" list="values" /><br />
<label for="temp">Temperature for room 2:</label><br />
<input type="range" id="temp" name="temp" list="values" /><br />
<label for="temp">Temperature for room 3:</label><br />
<input type="range" id="temp" name="temp" list="values" />

<datalist id="values">

>   <option value="0" label="0"></option>
>   <option value="25" label="0"></option>
>   <option value="50" label="0"></option>
>   <option value="75" label="0"></option>
>   <option value="100" label="0"></option>

Reviewer: ```suggestion
  <option value="0" label="0"></option>
  <option value="25" label="25"></option>
  <option value="50" label="50"></option>
  <option value="75" label="75"></option>
  <option value="100" label="100"></option>
```

Author: No I feel foolish

</datalist>
```

#### Result

{{EmbedLiveSample("Using the same datalist for multiple range controls")}}

### Adding labels

> You can label tick marks by giving the `<option>` elements `label` attributes. However, you must use CSS to show the labels and to position them correctly. Here's one way you could do this.

Reviewer: ```suggestion
You can label tick marks by giving the `<option>` elements `label` attributes. However, the label content will not be displayed by default. You can use CSS to show the labels and to position them correctly. Here's one way you could do this.
```

<input type="range" min="0" max="11" value="7" step="1" orient="vertical" />
```

> We target just the inputs with a type of range & orient of vertical, changing the writing mode from the default to `bt-lr`, or bottom-to-top and left-to-right, for Edge and Internet Explorer, and add `appearance: slider-vertical` which is supported in Blink and Webkit browsers:

Reviewer: ```suggestion
We target just the `input`s with a `type` of `range` and `orient` set to `vertical`, changing the `writing-mode` from the default to `bt-lr`, or bottom-to-top and left-to-right, for pre-Blink versions of Edge, and add `appearance: slider-vertical` which is supported in Blink and Webkit browsers:
```
